hooks/pre/31-upgrade-puppet.rb

```diff
@@ -1,20 +1,26 @@
-def stop_services                                                                                                                                                                                                                                     [0/1892]
+def puppet4_available?
+  success = []
+  success << Kafo::Helpers.execute('yum info puppet-agent')
+  success << Kafo::Helpers.execute('yum info puppetserver')
+  success.include?(false)
+end
+
+def stop_services
   Kafo::Helpers.execute('katello-service stop')
 end
 
-def upgrade_puppet_pkg
+def upgrade_puppet_package
   Kafo::Helpers.execute('yum remove -y puppet-server')
   Kafo::Helpers.execute('yum install -y puppetserver')
   Kafo::Helpers.execute('yum install -y puppet-agent')
 end
 
-def populate_p4_environments
-  Kafo::Helpers.execute('cp -rfp /etc/puppet/environments/* /etc/puppetlabs/code/environments')
-end
-
-def copy_ssl_certs
-  Kafo::Helpers.execute('mv /var/lib/puppet/ssl /etc/puppetlabs/puppet')
-  Kafo::Helpers.execute('mv /var/lib/puppet/foreman_cache_data /opt/puppetlabs/puppet/cache/')
+def copy_data
```

Comment:
I hate to be bikeshedding here, but given there's a move, should it just be ```migrate_files```? It also raises the question if we should move it all or copy it all instead of copy manifests but move certificates and caches.